### ISSUE TITLE
Ignore dotfiles directories by default

### ISSUE THREAD

**Issue by tkoft** [bndr/pipreqs#420]
I was hitting major issues because of the .venv and .mypy directories in my project. 

- The common issue with virtual environments is [already covered in this issue](https://github.com/bndr/pipreqs/issues/214), but ignoring dot-prefixed directories would fix this for 80% of people.
- Mypy stores cache files in directories named by the package being analyzed, so pipreqs thinks they are local packages and does not register them as external dependencies.

I can't think of a reason why source files should be in a dot directory, but if that's not a surefire rule, an alternative could be to copy [the default ignored paths that `isort` uses](https://pycqa.github.io/isort/docs/configuration/options.html#skip) ([source](https://github.com/PyCQA/isort/blob/main/isort/settings.py#L72-L94)):
```
DEFAULT_SKIP: FrozenSet[str] = frozenset(
    {
        ".venv",
        "venv",
        ".tox",
        ".eggs",
        ".git",
        ".hg",
        ".mypy_cache",
        ".nox",
        ".svn",
        ".bzr",
        "_build",
        "buck-out",
        "build",
        "dist",
        ".pants.d",
        ".direnv",
        "node_modules",
        "__pypackages__",
        ".pytype",
    }
)
```

I'd suggest taking a page from their book on [recognizing modules types](https://github.com/PyCQA/isort/blob/main/isort/place.py#L15) as well. Just pass in the module name, and ignore anything that's not "THIRDPARTY". It's not 100%, but it's tried and mostly true.

**Comment by doug-rehg** [comment]
I totally agree with this change request.    Either more robust exclusion, as described by @tkoft or a flag option to exclude directories is needed.
